src/preload/syscall_hook.S

```diff
@@ -572,6 +572,440 @@ SYSCALLHOOK_START(_syscall_hook_trampoline_49_89_ca)
+        .cfi_rel_offset x4, 80 + 8
+        stp     x5, x6, [sp, 96]
+        .cfi_rel_offset x3, 96
+        .cfi_rel_offset x4, 96 + 8
+        stp     x7, x9, [sp, 112]
+        .cfi_rel_offset x7, 112
+        .cfi_rel_offset x9, 112 + 8
+        stp     x10, x11, [sp, 128]
+        .cfi_rel_offset x10, 128
+        .cfi_rel_offset x11, 128 + 8
+        stp     x12, x13, [sp, 144]
+        .cfi_rel_offset x12, 144
+        .cfi_rel_offset x13, 144 + 8
+        stp     x14, x15, [sp, 160]
+        .cfi_rel_offset x14, 160
+        .cfi_rel_offset x15, 160 + 8
+        stp     x16, x17, [sp, 176]
+        .cfi_rel_offset x16, 176
+        .cfi_rel_offset x17, 176 + 8
+        str     x18, [sp, 192]
+        .cfi_rel_offset x18, 192
+        stp     q2, q3, [sp, 208]
+        .cfi_rel_offset q2, 208
+        .cfi_rel_offset q3, 208 + 16
+        stp     q4, q5, [sp, 240]
+        .cfi_rel_offset q4, 240
+        .cfi_rel_offset q5, 240 + 16
+        stp     q6, q7, [sp, 272]
+        .cfi_rel_offset q6, 272
+        .cfi_rel_offset q7, 272 + 16
+        // function call only maintain the bottom half of v8-v15
+        // whereas syscall maintains all the v registers
+        // so we actually need to save and restore v8-v15 as well...
+        stp     q8, q9, [sp, 304]
+        .cfi_rel_offset q8, 304
+        .cfi_rel_offset q9, 304 + 16
+        stp     q10, q11, [sp, 336]
+        .cfi_rel_offset q10, 336
+        .cfi_rel_offset q11, 336 + 16
+        stp     q12, q13, [sp, 368]
+        .cfi_rel_offset q12, 368
+        .cfi_rel_offset q13, 368 + 16
+        stp     q14, q15, [sp, 400]
+        .cfi_rel_offset q14, 400
+        .cfi_rel_offset q15, 400 + 16
+        stp     q16, q17, [sp, 432]
+        .cfi_rel_offset q16, 432
+        .cfi_rel_offset q17, 432 + 16
+        stp     q18, q19, [sp, 464]
+        .cfi_rel_offset q18, 464
+        .cfi_rel_offset q19, 464 + 16
+        stp     q20, q21, [sp, 496]
+        .cfi_rel_offset q20, 496
+        .cfi_rel_offset q21, 496 + 16
+        stp     q22, q23, [sp, 528]
+        .cfi_rel_offset q22, 528
+        .cfi_rel_offset q23, 528 + 16
+        stp     q24, q25, [sp, 560]
+        .cfi_rel_offset q24, 560
+        .cfi_rel_offset q25, 560 + 16
+        stp     q26, q27, [sp, 592]
+        .cfi_rel_offset q26, 592
+        .cfi_rel_offset q27, 592 + 16
+        stp     q28, q29, [sp, 624]
+        .cfi_rel_offset q26, 624
+        .cfi_rel_offset q27, 624 + 16
+        stp     q30, q31, [sp, 656]
+        .cfi_rel_offset q26, 656
+        .cfi_rel_offset q27, 656 + 16
+
+        bl      syscall_hook
+
+        movz    x29, #:abs_g1:alt_stack_nesting_level // assume 32bit address
+        movk    x29, #:abs_g0_nc:alt_stack_nesting_level
+        ldr     w30, [x29]
+        sub     w30, w30, 1
+        str     w30, [x29]
+
+        ldp     x29, x30, [sp, 32]
+        .cfi_restore x29
+        .cfi_restore x30
+        ldr     x8, [sp, 48]
+        .cfi_restore x8
```

Comment:
IIRC the default CFI for x0-x15 is cfi_undefined, so these should probably be `cfi_same_value`.

Reply:
Sure that works. I think gdb at least has a comment essentially saying that it'll treat absence of unwind info as same_value since gcc has/had a habbit of doing that..... I guess not every unwinder has that hack....

Reply:
I assume to be consistent I would also need to prefix other functions with a bunch of `.cfi_same_value`s though... I think this would even include most of the ASIMD registers.

Reply:
Yeah, that does sound annoying. I guess that's supposed to be done in the CIE, but it doesn't really let you encode those directly.